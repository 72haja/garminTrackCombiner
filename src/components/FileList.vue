<script setup>
const props = defineProps({
  localFiles: {
    type: Array,
    default: () => []
  }
})

const emit = defineEmits(['remove-file'])

function removeFile (file) {
  emit("remove-file", file)
}
</script>

<template>
  <div class="files-wrapper">
    <v-list v-for="file in localFiles"
            class="file-wrapper d-flex justify-space-between">
      <v-list-item class="file-name">{{ file.name }}</v-list-item>
      <v-list-item-action>
        <v-btn icon
               flat
               @click="removeFile(file)">
          <v-icon large
                  color="grey darken-2">
            mdi-close
          </v-icon>
        </v-btn>
      </v-list-item-action>
    </v-list>
  </div>
</template>
<style lang='scss'>
</style>
