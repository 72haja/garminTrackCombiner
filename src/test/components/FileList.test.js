import { describe, expect, vi } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'

import { createVuetify } from 'vuetify'

import FileList from '../../components/FileList.vue'

const testFile1 = new File(["Foo"], "testFile1", {
  type: "text/xml",
})
const testFile2 = new File(["Foo"], "testFile2", {
  type: "text/xml",
})

const tmpLocalFiles = [
  testFile1,
  testFile2,
]

describe('mount App', () => {
  const vuetify = createVuetify()

  test('mount component', async () => {
    expect(FileList).toBeTruthy()

    const wrapper = mount(FileList, {
      global: {
        plugins: [vuetify]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})


describe('test function', () => {
  const vuetify = createVuetify()
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(FileList, {
      props: {
        localFiles: tmpLocalFiles,
      },
      global: {
        plugins: [vuetify]
      }
    });
  });

  describe('removeFile', () => {
    test('removeFile', async () => {
      wrapper.vm.removeFile(testFile1)

      await wrapper.vm.$nextTick();

      expect(wrapper.emitted('remove-file')).toBeTruthy();
      expect(wrapper.emitted('remove-file')[0]).toEqual([
        testFile1,
      ]);
    })
  })
})

