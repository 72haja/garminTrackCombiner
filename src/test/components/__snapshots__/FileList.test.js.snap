// Vitest Snapshot v1

exports[`mount App > mount component 1`] = `"<div class=\\"files-wrapper\\"></div>"`;
